drivers/bridge/bridge.go

```diff
@@ -1451,3 +1479,60 @@ func electMacAddress(epConfig *endpointConfiguration, ip net.IP) net.HardwareAdd
 	}
 	return netutils.GenerateMACFromIP(ip)
 }
+
+func (d *driver) RestoreBridgeEndpoint(n *bridgeNetwork, endpoint *bridgeEndpoint, iface driverapi.InterfaceInfo) error {
+	ipv4Addr := iface.Address()
+	// Restore ipv4
+	_, err := ipAllocator.RequestIP(n.bridge.bridgeIPv4, ipv4Addr.IP)
+	if err != nil {
+		return err
+	}
+
+	n.Lock()
+	config := n.config
+	n.Unlock()
+
+	// Restore ipv6
+	ipv6Addr := iface.AddressIPv6()
+	if ipv6Addr.IP != nil && len(ipv6Addr.IP) != 0 {
+		network := n.bridge.bridgeIPv6
+		if config.FixedCIDRv6 != nil {
+			network = config.FixedCIDRv6
+		}
+		_, err := ipAllocator.RequestIP(network, ipv6Addr.IP)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Restore container interface name
+	sbox, err := linkByMacAddress(iface.MacAddress().String())
+	if err != nil {
+		return err
```

Comment:
@chenchun am wondering how you are testing these changes ? am facing all sorts of issues when I integrated it with docker daemon. am a bit concerned now that this PR is trying to do too many things.
Also, it would be good if you can explain the use-case that you are trying to address with this one. I remember you mentioning about the hot-restart of the docker daemon. But AFAIK it is not targetted for 1.9 release. So, it would be good if you can clarify the testing methodology and the use-case clearly asap.